tensor2tensor/data_generators/librispeech.py

```diff
@@ -69,41 +89,233 @@ def _get_audio_data(filepath):
   
   data = np.fromstring(byte_array, np.uint8).tolist()
   return data, frame_count, wav_file.getsampwidth(), wav_file.getnchannels()
-   
-
-def librispeech_generator(data_dir, tmp_dir, training, eos_list=None, start_from=0, how_many=0):
-  eos_list = [1] if eos_list is None else eos_list
-  datasets = (_LIBRISPEECH_TRAIN_DATASETS if training else _LIBRISPEECH_TEST_DATASETS)
-  i = 0
-  for url, subdir in datasets:
-    filename = os.path.basename(url)
-    compressed_file = generator_utils.maybe_download(tmp_dir, filename, url)
-
-    read_type = "r:gz" if filename.endswith("tgz") else "r"
-    with tarfile.open(compressed_file, read_type) as corpus_tar:
-      # Create a subset of files that don't already exist.
-      #   tarfile.extractall errors when encountering an existing file
-      #   and tarfile.extract is extremely slow
-      members = []
-      for f in corpus_tar:
-        if not os.path.isfile(os.path.join(tmp_dir, f.name)):
-          members.append(f)
-      corpus_tar.extractall(tmp_dir, members=members)
+      
+
+class LibrispeechTextEncoder(text_encoder.TextEncoder):
+
+  def encode(self, s):
+    return [ord[c] for c in s]

```

Comment:
should inlcued `self._num_reserved_ids `

Reply:
ord[c] is wrong.

Reply:
why not encode like self._num_reserved_ids + i ?

Reply:
Ah, good catch, I'll fix that syntax.  Regarding, num_reserved_ids, I based it on the timit generator:

https://github.com/tensorflow/tensor2tensor/blob/master/tensor2tensor/data_generators/audio.py#L150

which doesn't offset for num_reserved_ids, but you're right that it makes sense, so I will do so here.